### ISSUE TITLE
Polygon2D polygon data member can only be updated when you assign a new polygon

### ISSUE THREAD

**Issue by chucklepie** [godotengine/godot#42851]
**Godot version:**
3.2.2 stable

**OS/device including version:**
linux

**Issue description:**
A Polygon2D has a polygon property representing it. If you try to use the .set() method, the .remove() method or the .append() method on a Polygon2Ds polygon, absolutely nothing changes. 

If you change the values in the debug window, then the polygon is updated ok.

However, if you create a new PoolVector2Array, then assign it to the polygon it changes fine.

This may have something to do with PoolVector2Array being passed by value, but if it were, then you would either think why have the set and other methods, why is it not documented, and why are there no helper methods of Polygon2D to make changes more obvious...

Please see below for it not working and the code to make it work. Note, this class is missing a shed load of documentation and a lot of the properties (I have no idea about) are flagged red.

```
func _ready_shard():
	width=shard_texture.get_width()
	height=shard_texture.get_height()
	
	#NOTE: image_shard refers to the Polygon2D node in the scene

	#this updates the image fine
	image_shard.texture=shard_texture
	
	#this does not set the value, it remains as it was :(
	image_shard.polygon.set(0,Vector2(0,0))
	image_shard.polygon.set(1,Vector2(width,0))
	image_shard.polygon.set(2,Vector2(width,height))
	image_shard.polygon.set(3,Vector2(0,height))
	
	#this does not remove anything :(
	image_shard.polygon.remove(0)
	image_shard.polygon.remove(1)
	image_shard.polygon.remove(2)
	image_shard.polygon.remove(3)
	
	#this does not add anything :(
	image_shard.polygon.append(Vector2(0,0))	
	image_shard.polygon.append(Vector2(width,0))	
	image_shard.polygon.append(Vector2(width,height))	
	image_shard.polygon.append(Vector2(0,height))	

	#this works :)
	var a=[Vector2(0,0),Vector2(width,0),
	           Vector2(width,height),Vector2(0,height)]
	var p=PoolVector2Array(a)
	image_shard.polygon=p
```
**Steps to reproduce:**

**Minimal reproduction project:**

**Comment by akien-mga** [comment]
Duplicate of #15265.

**Comment by chucklepie** [comment]
Why close this? the case you're referring to is Closed also and refers to the append method inside dictionaries. Or does this mean you're happy with this inability to modify a Polygon2D polygon data usefully and without any documentation to indicate how it's done?

**Comment by akien-mga** [comment]
The issue I linked is not closed.

Polygon2D is not relevant here, the issue is that `polygon` is a `Pool*Array` and `Pool*Array` types can't be mutated when they're accessed as members. You need to assign a new `Pool*Array` instance to the `polygon` property to mutate it, as you found yourself in the OP.

**Comment by chucklepie** [comment]
Sorry, I just sometimes can't figure out the nesting and threading of the github. Reading all the way down this duplicate says 'Closed' :)

I know how to do this now, but anyone else it'll be the same problem. In the meantime can it be suggested somebody update the documentation, at least for Pool array to say if this is part of another node it won't work as expected)? I'd like to but I don't really know the Godot workflow.

**Comment by akien-mga** [comment]
> Sorry, I just sometimes can't figure out the nesting and threading of the github. Reading all the way down this duplicate says 'Closed' :)

What you might be seeing is other duplicate issues which have all been closed as duplicates of #15265.

#15265 itself is open:
![Screenshot_20201016_162929](https://user-images.githubusercontent.com/4701338/96271111-c6eb8180-0fcc-11eb-81d0-ddb01d7ae968.png)

It's a known limitation, and yes, until it's properly fixed in 4.0, it would be worth documenting in the description of all `Pool*Array` types.
